lib/stages/basebandReadout.cpp

```diff
@@ -163,8 +181,14 @@ void basebandReadout::main_thread() {
 
         frame_id++;
     }
-    if (mgr) {
-        mgr->stop();
+    if (mgrs[0]) {
+        //mgr->stop();
+        for(int freqidx = 0; freqidx < _num_local_freq; freqidx++){
+            if (mgrs[freqidx]){
+                mgrs[freqidx]->stop(); // stop all the managers that were started
```

Comment:
With can segfault if `mgrs[freqidx]` is an invalid pointer that is not null.

Reply:
Doesn't the typing protect you? It's a pointer to `basebandReadoutManager`, so you can't assign anything to it except for a readout manager. (Unless what you assign were freed.)

Which leads me to an important question: where exactly are these created?

Reply:
(Unless you're referring to `if (mgr[0])`, which is indeed incorrect and will segfault if `mgr`itself is null. And I don't know if it's guaranteed to be zeroed when the array is created, and so perhaps can be an invalid value until something is assigned.)

And to answer my own question, it's created in (current) line 156. But I don't like the way this is done with arrays -- why not a vector so it handles the empty case cleanly and can't have invalid pointers.